webapp/src/components/SingleStudy/MapView/index.tsx

```diff
@@ -0,0 +1,391 @@
+      },
+    },
+    graphView: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      width: '100%',
+      height: '100%',
+    },
+    areaCount: {
+      display: 'flex',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      width: '12%',
+      position: 'absolute',
+      right: theme.spacing(2),
+    },
+  }));
+
+interface Props {
+    study: StudyMetadata;
+}
+
+const FONT_SIZE = 15;
+const NODE_HEIGHT = 400;
+
+const calculateSize = (text: string): number => {
+  const textSize = text.length;
+  if (textSize === 1) {
+    return FONT_SIZE * textSize * 36;
+  }
+  if (textSize <= 2) {
+    return FONT_SIZE * textSize * 20;
+  }
+  if (textSize <= 3) {
+    return FONT_SIZE * textSize * 12;
+  }
+  if (textSize <= 5) {
+    return FONT_SIZE * textSize * 10;
+  }
+  if (textSize <= 6) {
+    return FONT_SIZE * textSize * 8.5;
+  }
+  if (textSize <= 10) {
+    return FONT_SIZE * textSize * 7.5;
+  }
+  return FONT_SIZE * textSize * 6.5;
+};
+
+const GraphViewMemo = React.memo(GraphView);
+
+const MapView = (props: Props) => {
+  const classes = useStyles();
+  const [t] = useTranslation();
+  const { study } = props;
+  const [loaded, setLoaded] = useState(false);
+  const [selectedItem, setSelectedItem] = useState<NodeProperties | LinkProperties>();
+  const [selectedNodeLinks, setSelectedNodeLinks] = useState<Array<LinkProperties>>();
+  const [nodeData, setNodeData] = useState<Array<NodeProperties>>([]);
+  const [linkData, setLinkData] = useState<Array<LinkProperties>>([]);
+  const { enqueueSnackbar } = useSnackbar();
+  const [openModal, setOpenModal] = useState<boolean>(false);
+  const [firstNode, setFirstNode] = useState<string>();
+  const [secondNode, setSecondNode] = useState<string>();
+  const graphRef = useRef<Graph<GraphNode & NodeProperties, GraphLink & LinkProperties>>(null);
+  const prevselectedItemId = useRef<string>();
+  // const [zoom, setZoom] = useState<number>();
+
+  const onClickNode = useCallback((nodeId: string) => {
+    if (!firstNode && nodeData) {
+      const obj = nodeData.find((o) => o.id === nodeId);
+      setSelectedItem(obj);
+    } else if (firstNode) {
+      setSecondNode(nodeId);
+    }
+  }, [firstNode, nodeData]);
+
+  const onClickLink = useCallback((source: string, target: string) => {
+    const obj = {
+      source,
+      target,
+    };
+    setSelectedItem(obj);
+  }, []);
+
+  const createModeLink = useCallback((id: string) => {
+    if (firstNode && firstNode === id) {
+      setFirstNode(undefined);
+      setSecondNode(undefined);
+    } else {
+      setFirstNode(id);
+    }
+  }, [firstNode, setFirstNode, setSecondNode]);
+
+  const onClose = () => {
+    setOpenModal(false);
+  };
+
+  const onSave = async (name: string, posX: number, posY: number, color: string) => {
+    setOpenModal(false);
+    try {
+      await createArea(study.id, name);
```

Comment:
this will return a AreaInfoDTO (see back), you should use the "id" field which can be different from the name field.
also NodeProperties should have an additionnal property "name" to put the name in it and display it in the NodeView instead of the id.